### DISCUSSION TITLE
Hardened Thunderbird Android!

### DISCUSSION THREAD

**Discussion by boredsquirrel** [dngray/ghacks-thunderbird-user.js#62] [Ideas]
Thunderbird Android is a completely separated project. It is based off the AOSP Mail app, which was continued in the fork "K9-Mail" and got integrated into the Thunderbird project.

The app is thus not based on Firefox or Gecko, and uses no user.js and a different way to store configs.

Still, I am sure this community would be interested in a hardening project. It will be a lot of maintenance, a result would be something like Mull, which is hardened Firefox Android.

Automatically patched codebase, APKs build in CI. 

I would like to learn how to do this, and would love to discuss about such a project!

**Comment by HorlogeSkynet** [comment]
Hello @boredsquirrel ! So now I have extensively browsed K9-Mail/Android TB settings, here is the list of privacy/security-related settings I've noticed :

* General > Display > Show correspondent name
* General > Privacy > Hide mail client (see #60) and timezone
* Per account > Reading mail > Show remote images
* (for IMAP) Per account > Fetching mail > Incoming server > Send client information
* Per account > End-to-End encryption

Are you aware of "hidden" settings that should be manually enforced ?
If not, I personally think those 3/4 settings are very easily to find and "intuitively" set to the required value.

Anyway, if your goal is to debloat (is K9 bloated ?), harden and recompile it like Mull does, such a project definitely requires a dedicated repository and maintainer(s).

Bye 👋

**Reply by boredsquirrel** [reply]
Hi there. I will have to look at it.

I am aware of the GUI settings and they might already make it pretty private. But there will be more things for people that want extreme privacy, like timezone, language etc.

Also looking at hardening the code would be interesting. There might be things that can be improved.

That sounds highly technical and kinda vague though. So until we find something worth fixing, I wouldnt invest effort in it :)

**Comment by lucasmz-dev** [comment]
Pretty sure K9/Thunderbird uses the system webview for showing e-mail content btw. It failed when I had it disabled (accidentally).

**Reply by HorlogeSkynet** [reply]
Hello @lucasmz-dev, even for plain text e-mails ? That would seem surprising though. Did you try with Bromite set as default ?
Thanks, bye :wave:

**Reply by lucasmz-dev** [reply]
When I have no WebView set (after a restart), everything seems to work except getting inside an email. Doesn't seem to matter which... So that's me extracting a "fact" out of a side-effect, makes sense to me though /shrug

I wouldn't use Bromite, Bromite is terribly out of date by now, and would be a bad idea IMO. Cromite has the browser but I don't think its webview is as easily installable in this case or cared for as much? IIRC it has the same package name as the system one so ehhh. 

I use CalyxOS, it works fine with its one, and also with Mulch System WebView which is whitelisted for use by CalyxOS as well, and a few others, like I believe Google's as well if someone really needs that for some reason.

Edit: Bromite isn't even whitelisted for use in Calyx anymore (if it even was at some point)

**Comment by HorlogeSkynet** [comment]
Interesting related thread : <https://forum.k9mail.app/t/hardening-k9-done-already/7796>.

**Reply by lucasmz-dev** [reply]
I feel like a lot of the thunderbird hardening is more relevant to the fact that Thunderbird for desktops is Firefox, a while browser forked to be an email client, while Thunderbird for Android is just K9, which is a native client. The displaying content is the more scary part, but they're using WebView, as long as that's properly restricted, it sounds fine though. You shouldn't be able to run JavaScript or WASM for example.

One thing that Thunderbird could add to Android (which is possible on PC with extensions) is a DKIM verifier, I know of some popular email servers people use that have improper DMARC tests atm.

**Comment by atomGit** [comment]
@lucasmz-dev - this is terribly off-topic, but you mentioned you use CalyxOS - i know virtually nothing about it, but people more knowledgeable than i are recommending against it and, personally, i don't like their app choices (Aurora, Chromium, MicroG, F-Droid - even Organic Maps [isn't a great choice](https://en.wikipedia.org/wiki/Organic_Maps#History) (see [CoMaps](https://comaps.app/) instead))

* [CalyxOS Is No Longer Safe to Use](https://www.youtube.com/watch?v=oTp8cTPH8g4)
* [Okay, nobody is asking so I will. What is happening? : CalyxOS](https://old.reddit.com/r/CalyxOS/comments/1mg6jdj/okay_nobody_is_asking_so_i_will_what_is_happening/)

**Reply by lucasmz-dev** [reply]
It is very off topic, but I would be careful with anything coming from Graphene side (that video comes from a guy that has weird tendencies around recommending  whatever they do purely) of the Android community. I never used Calyx because of Nick or Chirayu, and I'm awaiting its return ATM, but currently on iodé. It was the guys who actually developed the system in the end, even if some of them have now left.
I use Calyx as a Brazilian because it is the only decent OS for its type that is available for devices here

**Reply by lucasmz-dev** [reply]
The CoMaps thing is more a question of time... It is relatively new and by the time of the pause it was very new, the news were out but Organic was still the main one.

Also Chromium is their own fork, just not rebranded lol

**Comment by HorlogeSkynet** [comment]
Hello everyone, I'm closing this discussion in favor of this brand new wiki page : https://github.com/HorlogeSkynet/thunderbird-user.js/wiki/5.1.1-Thunderbird-Android
Thanks for your contributions and ideas, bye :wave:
